akka-actor/src/main/scala/akka/pattern/RetrySupport.scala

```diff
@@ -251,6 +249,21 @@ trait RetrySupport {
       delayFunction: Int => Option[FiniteDuration])(implicit ec: ExecutionContext, scheduler: Scheduler): Future[T] = {
     RetrySupport.retry(attempt, attempts, delayFunction, attempted = 0, shouldRetry)
   }
+
+  def retry[T](attempt: () => Future[T], retrySettings: RetrySettings)(
```

Comment:
Maybe: Feels natural that it'd be a separate parameter list in Scala, for this kind of usage:

```
retry(settings) { () => do it }
```

Reply:
I flipped the order, but it's not consistent with existing methods.

Reply:
Yeah, that's a good argument against, in streams which are newer than the APIs here we have a few that are more like my suggestion here, https://github.com/akka/akka/blob/3c1e5d23f68393d4a3c2ac276e5aa3a2bacbfd8a/akka-stream/src/main/scala/akka/stream/scaladsl/RestartSource.scala#L96

Look at that btw, _another_ case of pre-existing restart settings :)

Reply:
I think it's ok to flip the ordering for this specific case even if now aligned with the other methods. 

It's a much better API for Scala (like in streams)

Reply:
ok, done